src/function/built_in_vector_functions.cpp

```diff
@@ -103,7 +111,10 @@ uint32_t BuiltInVectorFunctions::getCastCost(
 
 uint32_t BuiltInVectorFunctions::getTargetTypeCost(LogicalTypeID typeID) {
     switch (typeID) {
-    case LogicalTypeID::INT16: {
+    case LogicalTypeID::INT16:
+    case LogicalTypeID::UINT64:
+    case LogicalTypeID::UINT32:
+    case LogicalTypeID::UINT16: {
```

Comment:
Should we add INT8 there as well?

Reply:
this cost is the cost for upward casting (i.e. from INT16 to INT32, but not the reverse). Since INT8 is the smallest, nothing will upward-cast to it. so we don't need INT8 here.